src/renault_api/kamereon/models.py

```diff
@@ -370,6 +370,45 @@ def for_json(self) -> Dict[str, Any]:
         return result
 
 
+@dataclass
+class HvacDaySchedule(BaseModel):
+    """Kamereon vehicle hvac schedule for day."""
+
+    readyAtTime: Optional[str]  # noqa: N815
+
+    def for_json(self) -> Dict[str, Any]:
+        """Create dict for json."""
+        return {
+            "readyAtTime": self.readyAtTime,
+        }
+
+
+@dataclass
+class HvacSchedule(BaseModel):
+    """Kamereon vehicle hvac schedule for week."""
+
+    id: Optional[int]
+    activated: Optional[bool]
+    monday: Optional[HvacDaySchedule]
+    tuesday: Optional[HvacDaySchedule]
+    wednesday: Optional[HvacDaySchedule]
+    thursday: Optional[HvacDaySchedule]
+    friday: Optional[HvacDaySchedule]
+    saturday: Optional[HvacDaySchedule]
+    sunday: Optional[HvacDaySchedule]
+
+    def for_json(self) -> Dict[str, Any]:
+        """Create dict for json."""
+        ret = {
+            "id": self.id,
+            "activated": self.activated,
+        }
+        for day in helpers.DAYS_OF_WEEK:
+            if self.__dict__.get(day):
+                ret[day] = self.__dict__[day].for_json()
```

Comment:
Needs type hint
```suggestion
        for day in helpers.DAYS_OF_WEEK:
            schedule: Optional[HvacDaySchedule] = self.__dict__.get(day)
            if schedule:  # pragma: no branch
                result[day] = schedule.for_json()
```

Reply:
Ack for the type hint, sorry for that. However, I disagree on the suggestion: This would lead to "empty" days being present in the data sent to the API. For me, the API rejected this. By testing on presence of the days, this would send only the actually configured days — which the API accepts …

Reply:
I'm not sure what you mean. My suggestion has the same effect as your code:
- If the schedule exists, it is written to the json output.
- If the schedule doesn't exist, it is skipped.

Happy for you to provide another solution for type hint.

Reply:
Couldn’t read … my bad